### ISSUE TITLE
skipIfMatch skips full line; unexpectedly skips typos

### ISSUE THREAD

**Issue by DavidRieman** [aotaduy/eslint-plugin-spellcheck#30]
I've tried adding the documented sample URL regex to skipIfMatch to eslint.json, like so:
`"skipIfMatch": [
   "http://[^s]*"
]`
(and really it will usually be more like "https?://" to also skip secure URLs as well, in the end)

Consider the following JavaScript comment:
```
                // words, words, some more words... which are
                // docmented at: http://asdfasdfasdf.org
```

Unfortunately, although the URL is skipped correctly, the "docmented" typo is _accidentally not reported_.  It seems that if a skipIfMatch expression matches, then spell check is skipped for the entire line.

If intended, I recommend:
* Rename to "skipLineIfMatch"
* Document the behavior

If unintended, I recommend:
* Fix to apply expression per word
* Rename to "skipWordIfMatch"
* Document the behavior

Or, we can have both options, with the explicit names, and both with documented behavior.  (Although with explicit names it would be easier to intuit intent even in lieu of documentation.)

**Comment by DavidRieman** [comment]
Also note there are additional cases where we really want to ignore a swath of things like obvious UIDs of various sizes (I'm planning to ignore any word with a number in the middle) without ignoring spelling on the receiving variables, etc.

**Comment by DavidRieman** [comment]
Let me know if one of these approaches is likely to be well received.  I may be able to find time this weekend to fork and investigate, and hopefully implement a contribution.  :)

**Comment by aotaduy** [comment]
Hello David, the problem here is that skipIfMatch will skip the entire comment if the url is found in any part of it, be aware that will not skip the line, it will skip the js language node that eslint provides to the plugin. 

I'll document this on the configuration section as you suggested and will think in a solution, I think that your suggestion about adding a skipIfMatchWord parameter could be a good approach, but it will require a bit more work revising the word splitting logic of the plugin.

**Comment by aotaduy** [comment]
We added this feature so closing
